Implementation of VIFs for path coefficient and Mode B weights

FloSchuberth opened the pull request:
I have added the VIF for the path coefficients and for Mode B weights.

I have tested the following cases for the VIF of the path coefficients:
- simple regression, i.e., models with equations that contain only a single independent variable
- multiple regression, i.e., models with equations that contain more than one independent variable
This has been checked in case of estimating several models and if bootstrap is used (Error calculation method = Robust)

Moreover, I have tested the following cases for the VIF of the Mode B weights:
- single indicator composite
- no composite
- composite with more than one indicator
This has been checked in case of estimating several models (using a grouping variable) and if bootstrap is used.

juliuspfadt commented:
when I have the following jasp file 
[testVIF.jasp.zip](https://github.com/jasp-stats/jaspSem/files/14324151/testVIF.jasp.zip)
I see the VIF, but it is the equal for both predictors. Is that normal?

juliuspfadt reviewed (changes requested):
Thanks a lot Flo! It looks very good. For requesting changes see the comment with removing the empty column.

juliuspfadt commented:
Ah, I just saw that the added columns do mess with the unit tests. so maybe lets do it in the order: change the empty column behavior, and then I can fix the tests

FloSchuberth commented:
> when I have the following jasp file [testVIF.jasp.zip](https://github.com/jasp-stats/jaspSem/files/14324151/testVIF.jasp.zip) I see the VIF, but it is the equal for both predictors. Is that what you would expect?

Yes this is normal. In PLS-SEM the constructs are always standardized, i.e., they have mean zero and unit variance. Consequently, if you have only two independent variables (IVs) in an equation, the VIFs will be the same. The VIF is generally defined as 1/(1-R_j^2), where R_j^2 is the R^2 of the regression of the IV j on the other IVs: https://en.wikipedia.org/wiki/Variance_inflation_factor

FloSchuberth commented:
> Thanks a lot Flo! It looks very good. For requesting changes see the comment with removing the empty column and two more things:
> 
>     * we need at least one unit test for the VIFs in https://github.com/jasp-stats/jaspSem/blob/master/tests/testthat/test-plssem.R. If you dont feel comfortable doing it, I can also take over. If you are interested, how to do to, see this https://github.com/jasp-stats/jasp-desktop/blob/stable/Docs/development/r-unit-test-guide.md.
>       
>       * I usually start running the analysis as I want it in JASP, then I save the JASP file, and in R I run    `jaspTools::analysisOptions("file_path")`. Setting up jaspTools should be explained in the guide above.
>       * then I have R print the options with `dput` and I use them to run the analysis. You can look at how the other tests do it
>       * I think the most annoying thing is to specify the options
>       * when you do `jaspTools::runAnalysis()` with the argument `makeTests=TRUE` you get the printed code to put into the test file
> 
>     * we should also update the help files https://github.com/jasp-stats/jaspSem/blob/master/inst/help/PLSSEM.md to reflect the new changes. Obviously I think you would be the best person for the job, but if all this becomes too much I can also do it

@juliuspfadt: 
Considering the tests, I would be happy if you could do that. 

Considering the update of the help file, I can do that. However, it is not clear to me what exactly needs to be updated. There is no section about changes or updates, it is rather a general description that JASP is based on the cSEM package and some references. Considering the packages used nothing has changed. This is what I have done:
- Renamed Factor loading to Loading in the tbale as this table can contain both factor and composite loadings
- Changed Latent variable to construct in the table as the table does not always contain LV (e.g., if a composite is specified) so I thought construct is a more suitable expression  
- Implement the VIF for path coefficients
- Implement the VIF for mode B weights
- Changed the column order of table, i.e., VIF and f^2 values are reported at the end.

juliuspfadt commented:
You are right, the helpfile is a mess. It should follow a structure such as https://github.com/jasp-stats/jaspFactor/blob/master/inst/help/ConfirmatoryFactorAnalysis.md. I see that the whole SEM module is missing proper help files... well. I suppose we leave the help files for another PR. 

Do you feel comfortable implementing the empty column change?

FloSchuberth commented:
> You are right, the helpfile is a mess. It should follow a structure such as https://github.com/jasp-stats/jaspFactor/blob/master/inst/help/ConfirmatoryFactorAnalysis.md. I see that the whole SEM module is missing proper help files... well. I suppose we leave the help files for another PR.
> 
> Do you feel comfortable implementing the empty column change?

@juliuspfadt 
I have implemented the empty column change, see my latest commit. The VIF column is removed if it is empty, otherwise it is shown. 

To achieve that I had to 'mess up' a bit the original code structure. Originally, the table were created first, parameters were estimated and then the tables were filled. Now the vif columns are only added if there are vifs. So creating the columns for the vif is now in the section where the parameters are estimated and no longer in the section where the tables are created.

juliuspfadt commented on R/plssem.R:
```diff
@@ -576,7 +576,7 @@ checkCSemModel <- function(model, availableVars) {
   }
 
-  weightTab$addColumnInfo(name = "vifb",      title = gettext("VIF"),   type = "number")
+  # weightTab$addColumnInfo(name = "vifb",      title = gettext("VIF"),   type = "number")
```
remove code that is commented out

juliuspfadt commented on R/plssem.R:
```diff
@@ -624,7 +624,7 @@ checkCSemModel <- function(model, availableVars) {
   }
   pathTab$addColumnInfo(name = "f2",       title = "f\u00B2",               type = "number")
-  pathTab$addColumnInfo(name = "vif",      title = gettext("VIF")     ,     type = "number")
+  # pathTab$addColumnInfo(name = "vif",      title = gettext("VIF")     ,     type = "number")
```
same

juliuspfadt reviewed (approved):
Looks good. I will implement a unit test, and the merge. Thanks again!
